core/insertion_marker_manager.js

```diff
@@ -40,743 +40,746 @@ const {WorkspaceSvg} = goog.requireType('Blockly.WorkspaceSvg');
+   * Shows an outline around the input the closest connection belongs to.
+   * @private
+   */
+  showInsertionInputOutline_() {
+    const closest = this.closestConnection_;
+    this.highlightedBlock_ = closest.getSourceBlock();
+    this.highlightedBlock_.highlightShapeForInput(closest, true);
   }
 
-  this.markerConnection_ = null;
-  const svg = imBlock.getSvgRoot();
-  if (svg) {
-    svg.setAttribute('visibility', 'hidden');
+  /**
+   * Hides any visible input outlines.
+   * @private
+   */
+  hideInsertionInputOutline_() {
+    this.highlightedBlock_.highlightShapeForInput(this.closestConnection_, false);
+    this.highlightedBlock_ = null;
   }
-};
 
-/**
- * Shows an outline around the input the closest connection belongs to.
- * @private
- */
-InsertionMarkerManager.prototype.showInsertionInputOutline_ = function() {
-  const closest = this.closestConnection_;
-  this.highlightedBlock_ = closest.getSourceBlock();
-  this.highlightedBlock_.highlightShapeForInput(closest, true);
-};
+  /**
+   * Shows a replacement fade affect on the closest connection's target block
+   * (the block that is currently connected to it).
+   * @private
+   */
+  showReplacementFade_() {
+    this.fadedBlock_ = this.closestConnection_.targetBlock();
+    this.fadedBlock_.fadeForReplacement(true);
+  }
 
-/**
- * Hides any visible input outlines.
- * @private
- */
-InsertionMarkerManager.prototype.hideInsertionInputOutline_ = function() {
-  this.highlightedBlock_.highlightShapeForInput(this.closestConnection_, false);
-  this.highlightedBlock_ = null;
-};
+  /**
+   * Hides/Removes any visible fade affects.
+   * @private
+   */
+  hideReplacementFade_() {
+    this.fadedBlock_.fadeForReplacement(false);
+    this.fadedBlock_ = null;
+  }
 
-/**
- * Shows a replacement fade affect on the closest connection's target block
- * (the block that is currently connected to it).
- * @private
- */
-InsertionMarkerManager.prototype.showReplacementFade_ = function() {
-  this.fadedBlock_ = this.closestConnection_.targetBlock();
-  this.fadedBlock_.fadeForReplacement(true);
-};
+  /**
+   * Get a list of the insertion markers that currently exist.  Drags have 0, 1,
+   * or 2 insertion markers.
+   * @return {!Array<!BlockSvg>} A possibly empty list of insertion
+   *     marker blocks.
+   * @package
+   */
+  getInsertionMarkers() {
+    const result = [];
+    if (this.firstMarker_) {
+      result.push(this.firstMarker_);
+    }
+    if (this.lastMarker_) {
+      result.push(this.lastMarker_);
+    }
+    return result;
+  }
+}
 
 /**
- * Hides/Removes any visible fade affects.
- * @private
+ * An enum describing different kinds of previews the InsertionMarkerManager
+ * could display.
+ * @enum {number}
  */
-InsertionMarkerManager.prototype.hideReplacementFade_ = function() {
-  this.fadedBlock_.fadeForReplacement(false);
-  this.fadedBlock_ = null;
+InsertionMarkerManager.PREVIEW_TYPE = {
+  INSERTION_MARKER: 0,
+  INPUT_OUTLINE: 1,
+  REPLACEMENT_FADE: 2,
```

Comment:
This should probably be renamed `PreviewType`.  Also, I would either make it a `static` item in the `class` definition, or make it its own separate named export.